### ISSUE TITLE
[Feature] Auto-lock PR Conversations When Merged

### ISSUE THREAD

**Issue by kdumontnu** [go-gitea/gitea#22121]
### Feature Description

I think it would be helpful if repos had a setting to automatically lock a conversation when an issue is closed and/or PR is merged.

Related to: https://github.com/go-gitea/gitea/issues/20012

**Comment by delvh** [comment]
Wouldn't it be better to let a CI do that?
Then we don't have a weird edge case that is only sometimes wanted in our core while still providing the functionality…
As far as I know, the CI currently proposed should (later on) be able to do that (see #21937)

**Comment by kdumontnu** [comment]
> Wouldn't it be better to let a CI do that? Then we don't have a weird edge case that is only sometimes wanted in our core while still providing the functionality… As far as I know, the CI currently proposed should (later on) be able to do that (see #21937)

Yeah, I appreciate your point, but CI seems like overkill for this if a user isn't already running CI for something else (also, is there a webhook for issues closed or would the CI job it have to poll each open issue?). It seems like what  you suggest is the [strategy GitHub has taken](https://github.com/dessant/lock-threads), as there doesn't seem to be a feature like this built in. From a UX perspective, it seems like like a nice compliment to auto-merge.

If this really is a niche feature that a majority of users don't want/need, I agree it should go to CI. If a reasonable number of folks want this it would be nice to build into the UI. I'll post in discord and see if other people would like this feature, and if there's not much traction in a week then I'm perfectly happy closing this.

**Comment by jolheiser** [comment]
> also, is there a webhook for issues closed or would the CI job it have to poll each open issue?

Yes-ish, the base `issues` covers "Issue opened, closed, reopened, or edited."

**Comment by lunny** [comment]
> > Wouldn't it be better to let a CI do that? Then we don't have a weird edge case that is only sometimes wanted in our core while still providing the functionality… As far as I know, the CI currently proposed should (later on) be able to do that (see #21937)
> 
> Yeah, I appreciate your point, but CI seems like overkill for this if a user isn't already running CI for something else (also, is there a webhook for issues closed or would the CI job it have to poll each open issue?). It seems like what you suggest is the [strategy GitHub has taken](https://github.com/dessant/lock-threads), as there doesn't seem to be a feature like this built in. From a UX perspective, it seems like like a nice compliment to auto-merge.
> 
> If this really is a niche feature that a majority of users don't want/need, I agree it should go to CI. If a reasonable number of folks want this it would be nice to build into the UI. I'll post in discord and see if other people would like this feature, and if there's not much traction in a week then I'm perfectly happy closing this.

That Action could be reused when Gitea Actions PR merged.

**Comment by a1012112796** [comment]
How about add an extern option to do it ?
![屏幕截图 2022-12-14 170346](https://user-images.githubusercontent.com/25342410/207742844-35b07f34-2911-4896-a85d-5a1915eb2aa5.png)

**Comment by lunny** [comment]
Not a good idea to put more and more checkboxes/buttons there.
